index.js

```diff
@@ -9,15 +9,25 @@ export default class Flatbush {
 
     /**
      * Recreate a Flatbush index from raw `ArrayBuffer` or `SharedArrayBuffer` data.
-     * @param {ArrayBuffer | SharedArrayBuffer} data
+     * @param {Uint8Array | ArrayBuffer | SharedArrayBuffer} data
      * @returns {Flatbush} index
      */
     static from(data) {
-        // @ts-expect-error duck typing array buffers
-        if (!data || data.byteLength === undefined || data.buffer) {
-            throw new Error('Data must be an instance of ArrayBuffer or SharedArrayBuffer.');
+        let byteOffset = 0;
+        if (data instanceof Uint8Array) {
+            byteOffset = data.byteOffset;
+            if (byteOffset % 8 !== 0) {
+                throw new Error('Uint8Array offset must be 8-byte aligned.');
+            }
```

Comment:
Can you say more on why we need to validate byte alignment here and why `8` specifically? For data buffer size, we have: 

```
8 + // header
numNodes * 4 * ArrayType.BYTES_PER_ELEMENT +  // box values
numNodes * IndexArrayType.BYTES_PER_ELEMENT // index values
```

If `numNodes < 16384`, then the index values are Uint16Array — just two bytes per node.

Reply:
It's the creation of the view, relevant both for the indices TypedArray and for the boxes' TypedArray. For the indices, it'll be either a Uint16Array or a Uint32Array, which requires byte alignment of 2 or 4 bytes respectively. But if you use a Float64Array for the boxes, that needs to be 8-byte aligned. Ref https://stackoverflow.com/a/8638875.

If we remove this error check, testing with `byteOffset = 10` gives
```
  RangeError: start offset of Uint32Array should be a multiple of 4
      at new Uint32Array (<anonymous>)
      at Flatbush.from (file:///Users/kyle/github/mourner/flatbush/index.js:41:28)
```

Testing with `byteOffset = 12` gives
```
  RangeError: start offset of Float64Array should be a multiple of 8
      at new Float64Array (<anonymous>)
      at new Flatbush (file:///Users/kyle/github/mourner/flatbush/index.js:87:27)
```

So the options are:

- always check 8-byte alignment for simplicity
- improve this check to use the `ArrayType.BYTES_PER_ELEMENT` instead of a hard-coded 8
- Don't have a check and let the typed array constructor throw, and document to the user it's an issue with their input, not this library

Reply:
@kylebarron all right, let's keep the 8 byte check — we can always improve this later if someone runs into this.